js/scripts.js

```diff
@@ -0,0 +1,111 @@
+var carouselList = $("#carousel ul"),
+    slidesInterval = setInterval(changeSlide, 5000),
+    next = $("#next"),
+    prev = $("#prev");
+
+    //NEXT SLIDE
+
+function changeSlide() {
+    carouselList.animate({'marginLeft':-400}, 1000, moveFirstSlide);
+};
+
+function moveFirstSlide() {
+    var firstSlide = carouselList.find("li:first");
+    var lastSlide = carouselList.find("li:last");
+
+    lastSlide.after(firstSlide);
+    carouselList.css({marginLeft:0});
+};
+
+next.click(function(){
+    clearInterval(slidesInterval);
+    changeSlide();
+    slidesInterval = setInterval(changeSlide, 5000);
+});
+
+    //PREVIOUS SLIDE 
+
+function prevSlide() {
+    carouselList.css({marginLeft:-400});
+    moveLastSlide();
+    carouselList.animate({'marginLeft': 0}, 1000);
+};
+
+function moveLastSlide() {
+    var firstSlide = carouselList.find("li:first");
+    var lastSlide = carouselList.find("li:last");
+
+    firstSlide.before(lastSlide);
+};
+
+prev.click(function(){
+    clearInterval(slidesInterval);
+    prevSlide();
+    slidesInterval = setInterval(changeSlide, 5000);
+});
+
+    //CONTROLS
+
+var controlsInterval = setInterval(checkPosition, 100);
+
+function checkPosition() {
```

Comment:
better make this function with checking index of element - in jQuery is method `.index()`

Reply:
I'm not sure how - index of each slide changes all the time and the only thing that doesn't change is id. That's why function is checking id insted of index.

Reply:
yes, but you can attach an index to e.g. `data-index=""` and operate on it, remember that your function should be more flexible and reusable ;)